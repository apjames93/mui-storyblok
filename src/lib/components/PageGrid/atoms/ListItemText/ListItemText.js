import React from 'react';
import PropTypes from 'prop-types';
import { ListItemText as MuiListItemText } from '@material-ui/core';
import Storyblok from 'lib/utils/Storyblok';

/**
 * MuiListItemText is used in storyblok redirect to react routes
 */

export const ListItemText = ({
  secondary,
  primary,
  rootClass,
  storyblokClass,
  dataBlokC,
  dataBlokUid,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <MuiListItemText
      className={`${styles.root} ${storyblokClass}`}
      primary={primary}
      secondary={secondary}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
    />
  );
};

export default ListItemText;

ListItemText.propTypes = {
  /** primary text */
  primary: PropTypes.string.isRequired,
  /** secondary text */
  secondary: PropTypes.string,
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),
};

ListItemText.defaultProps = {
  rootClass: [],
  secondary: '',
};
