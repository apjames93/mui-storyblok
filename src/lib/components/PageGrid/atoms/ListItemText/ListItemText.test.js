import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListItemText from './ListItemText';

function setup() {
  const props = {
    component: 'ListItemText',
    primary: 'primary',
    secondary: 'secondary',
  };
  const comp = shallow(<ListItemText {...props} />);
  return { comp, props };
}

describe('<ListItemText />', () => {
  it('renders ListItemText', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<ListItemText {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
