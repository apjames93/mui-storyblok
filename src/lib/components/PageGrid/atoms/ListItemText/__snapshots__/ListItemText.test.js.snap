// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemText /> snapshot 1`] = `
<div
  className="MuiListItemText-root makeStyles-root-3 function () { [native code] } MuiListItemText-multiline"
  data-blok-c={[Function]}
  data-blok-uid={[Function]}
>
  <span
    className="MuiTypography-root MuiListItemText-primary MuiTypography-body1 MuiTypography-displayBlock"
  >
    primary
  </span>
  <p
    className="MuiTypography-root MuiListItemText-secondary MuiTypography-body2 MuiTypography-colorTextSecondary MuiTypography-displayBlock"
  >
    secondary
  </p>
</div>
`;
